### DISCUSSION TITLE
React Native dependency simplification

### DISCUSSION THREAD

**Discussion by jamiely** [WalletConnect/walletconnect-monorepo#698] [Ideas]
Thank you so much for your work on the Wallet Connect protocol. I’ve found it to be straightforward to use and implement; however the NodeJS polyfill requirement is a blocker for my use case.

I am working on a react-native mobile app with a large amount of existing code. I was able to add @walletconnect/react-native-dapp by polyfilling NodeJS dependencies using Babel. The requirement to polyfill the NodeJS libraries adds friction in the form of more complex configuration and additional dependencies that duplicate functionality already available, adding to application bundle size. For example, at a minimum, it seems like it is necessary to polyfill the http, https, and various stream-related modules.

I think that the polyfill requirement introduces barriers to adoption for those with large existing applications. As the adoption of web3 becomes more mainstream, it seems like this may increasingly become an issue. If I were to fork @walletconnect/react-native-app version 1 in order to simplify dependencies, would there be any interest in a contribution back upstream?

**Comment by jamiely** [comment]
I took a look at a bunch of the code, and it seems like reimplementing the crypto interface defined in the `@walletconnect/iso-crypto` package is the simplest thing to do to remove a bunch of other dependencies. Thoughts on this approach? What I like about this approach is there are only 4 functions to reimplement, and they are implementable using something like crypto-js. 

I can polyfill an alternative implementation as described above for now, but what are the thoughts on allowing someone to pass in this dependency in the [`WalletConnect` constructor](https://github.com/WalletConnect/walletconnect-monorepo/blob/c6b26481c34848dbc9c49bb0d024bda907ec4599/packages/clients/client/src/index.ts#L5)? It might also be worth allowing someone to pass in this option in the [`@walletconnect/react-native-dapp` package](https://github.com/WalletConnect/walletconnect-monorepo/blob/c6b26481c34848dbc9c49bb0d024bda907ec4599/packages/helpers/react-native-dapp/src/providers/WalletConnectProvider.tsx#L158)

**Comment by pedrouid** [comment]
Yeah we can easily make it pass through on @walletconnect/client

The reason for not using crypto-js was purely for performance

While NodeJS polyfills seem unideal they do use native crypto API binded to NodeJS API
